_posts/2024-XX-XX-quarkus-kiota.adoc

```diff
@@ -0,0 +1,168 @@
+---
+layout: post
+title: 'The road to generated SDKs with Kiota using Quarkus'
+date: 2024-XX-XX
+tags: release
+synopsis: 'Introducing a Quarkus-Kiota extension and available Java tools to work with Kiota.'
+author: andreaTP
+---
+
+== The Challenge
+:imagesdir: assets/images/posts/kiota
+
+Quarkus applications typically expose functionalities through API endpoints; here, we discuss how to easily consume those APIs and provide a smooth experience for our beloved users. While HTTP calls are easy to perform in any programming language and environment, when API complexity increases, and the number of available endpoints becomes uncountable, the need for better tools becomes apparent.
+
+image::api_monster.jpeg[API monster]
+
+=== Developer Experience Wishlist:
+
+- Fully typed programmatic APIs that make discovering and invoking HTTP endpoints safe, expressive, and elegant
+- An easy and idiomatic way to plug in various authentication mechanisms
+- Smooth upgrades of the generated code when the description evolves
+
+Looking at the learnings from big companies facing this issue at scale, a viable solution is to ship multiple language-specific SDKs (Software Development Kits), i.e. for Java, Python, Go, JavaScript etc. For years, companies have been paying developers to manually provide the abstraction layer, making cloud services functionalities more easily accessible. This approach adds up to the maintenance burden, increases the operational complexity of releases, and, worst, is tedious for developers.
+
+To pave the way for better tooling, new standards are being defined in the computer industry. For building HTTP-based APIs, link:https://www.openapis.org[OpenAPI] stands out as a widely adopted option.
+In the context of this article we are going to assume you already have an OpenAPI description of the service.
+
+=== Current Status in Quarkus
+
+Quarkus already provides an expressive and fully integrated link:https://quarkus.io/guides/rest-client-reactive[Rest Client] to enable you to craft HTTP calls beautifully.
+Additionally, the link:https://github.com/quarkiverse/quarkus-openapi-generator[quarkus-openapi-generator] is a mature Quarkus extension designed to generate client code from OpenAPI.
+This option is recommended when a tight integration with Quarkus is your primary focus.
+
+In this post, we will introduce an alternative solution that addresses the same problem but makes different trade-offs to prioritize consistency across languages and frameworks.
+
+=== A Story of Open Source Collaboration
+
+==== Microsoft Graph - Kiota
```

Comment:
```suggestion
==== Microsoft - Kiota
```

(this is how we've been introducing the tool to date, I'd make sure the name of the project is consistent, but keep the team name to outline the collaboration work)

Reply:
@maxandersen specifically asked to define the teams involved in this collaboration.
I'm happy with whatever you two agree on.

Reply:
changing the section title is fine if ok by Microsoft  - the specific origin is mentioned in the body which is fine.